### DISCUSSION TITLE
can i retrain on my custom dataset?

### DISCUSSION THREAD

**Discussion by pure-rgb** [mindee/doctr#977] [Q&A]
If so, how can I do that? I couldn't fine any resource in DocTR.

**Comment by frgfm** [comment]
Hello @Suzan009 :wave: 

For which task would you like to train a docTR model?
You can find readmes and training script for both Tensorflow and PyTorch for multiples tasks over here :point_down: https://github.com/mindee/doctr/tree/main/references

Let me know if the instructions are unclear but they should guide you to meet the required data format using your own dataset :+1: 

Cheers :v:

**Reply by pure-rgb** [reply]
@frgfm 

> For which task would you like to train a docTR model?

My task is to detect and recognize text from images. I want to re-train my custom synthetic datasets, followed by fine-tuning the downstream task.

> You can find readmes and training script for both Tensorflow and PyTorch for multiples tasks over here 👇 [main/references](https://github.com/mindee/doctr/tree/main/references)

Okay, did a quick look. It seems like I should be able to do this. But isn't the procedure documented? As you showed in the read-me file? Here are the things I am looking for, In order to train,

- How to prepare my dataset for the DocTR library? 
- How to set up config file (image size, other hyper-parameters, and data augmentation)?
- How to train the detection model?
- How to train the recognition model? 
- How do infer both of these models?

**Reply by frgfm** [reply]
I'm confused :sweat_smile: 

In the readme you can find the data format pour the training, the training script comes with a  `--help` flag with details for every single script argument. Then there is a readme for each task (meaning for both detection and recognition)

There is only the last part about how to perform inference with your custom model that isn't well documented, I'll try to fix this shortly :) 

In the meantime, feel free to ask anything that isn't clear!

**Reply by felixdittrich92** [reply]
@frgfm be careful with the next release this will be much easier :)
You can pass the model directly without some tricky wrappings

```
model = db_resnet50(pretrained=False)
model.load_state_dict('path-to-your-model.pt', map_location='cpu')
# only detection
detector = detection_predictor(model, pretrained=False)
# full pipe
predictor = ocr_predictor(det_arch=model, pretrained=True)
```

same with from_hub
```
model = from_hub(modelcard_id)
# full pipe
predictor = ocr_predictor(det_arch=model, pretrained=True)
```

same for TF

**Reply by frgfm** [reply]
@felixdittrich92 yup :) 
But the README of the references section always have the latest version available :thinking: 

What did you mean? (upcoming non-pushed changes)

**Reply by felixdittrich92** [reply]
You wrote that you will update a section how to load custom models :)  this will change or better be asier with the next release

So i want only to ensure that you not add something like this: :sweat_smile: 

```
det_predictor = DetectionPredictor(
        PreProcessor((1024, 1024), batch_size=1, mean=(0.798, 0.785, 0.772), std=(0.264, 0.2749, 0.287)), det_model)
    # cf. https://github.com/mindee/doctr/blob/main/references/recognition/train_pytorch.py#L97 & https://github.com/mindee/doctr/blob/main/references/recognition/train_pytorch.py#L111
    reco_predictor = RecognitionPredictor(
        PreProcessor((32, 128), preserve_aspect_ratio=True, batch_size=32, mean=(0.694, 0.695, 0.693),
                     std=(0.299, 0.296, 0.301)), reco_model)
    print("Time Consumed for model loading")
    print("% s seconds" % (time.time() - load_mdl_strt))
    start_mdl = time.time()
    model = OCRPredictor(det_predictor, reco_predictor)
```
